include/libopencm3/stm32/h7/usart.h

```diff
@@ -40,21 +41,151 @@
  * Holds all the U(S)ART peripherals supported.
  * @{
  */
-#define USART1          USART1_BASE
-#define USART2          USART2_BASE
-#define USART3          USART3_BASE
-#define UART4           UART4_BASE
-#define UART5           UART5_BASE
-#define USART6          USART6_BASE
-#define UART7           UART7_BASE
-#define UART8           UART8_BASE
+#define USART1 USART1_BASE
+#define USART2 USART2_BASE
+#define USART3 USART3_BASE
+#define UART4  UART4_BASE
+#define UART5  UART5_BASE
+#define USART6 USART6_BASE
+#define UART7  UART7_BASE
+#define UART8  UART8_BASE
+/**@}*/
+
+/** @addtogroup usart_registers
+@{*/
+#define USART_PRESC(usart_base) MMIO32((usart_base) + 0x2C)
+/**@}*/
+
+/** @addtogroup usart_cr1_values
+@{*//** RX FIFO Full Interrupt Enable. */
+#define USART_CR1_RXFFIE BIT31
+/** TX FIFO Empty Interrupt Enable. */
+#define USART_CR1_TXFEIE BIT30
+/** FIFO Enable bit. */
+#define USART_CR1_FIFOEN BIT29
+/**@}*/
+
+/** @addtogroup usart_cr3_values
+@{*/
+/** FIFO Threshold definitions. */
+typedef enum {
+	USART_FIFO_THRESH_EIGTH = 0x0,
+	USART_FIFO_THRESH_QUARTER = 0x1,
+	USART_FIFO_THRESH_HALF = 0x2,
+	USART_FIFO_THRESH_THREEQTR = 0x3,
+	USART_FIFO_THRESH_SEVENEIGTH = 0x4,
+	USART_FIFO_THRESH_TX_EMPTY = 0x5,
+	USART_FIFO_THRESH_RX_FULL = 0x5,
+	USART_FIFO_THRESH_MASK = 0x7
+} usart_fifo_threshold_t;
+
+/** TX FIFO Threshold Configuration bits. */
+#define USART_CR3_TXFTCFG_SHIFT 29
+/** RX FIFO THreshold Interrupt Enable. */
+#define USART_CR3_RXFTIE BIT28
+/** RX FIFO Threshold Configuration bits. */
+#define USART_CR3_RXFTCFG_SHIFT 25
+/** Transmission Complete Before Guard Time Enable bit. */
+#define USART_CR3_TCBGTIE BIT24
+/** TX FIFO THreshold Interrupt Enable. */
+#define USART_CR3_TXFTIE BIT23
+/**@}*/
+
+/** @addtogroup usart_isr_values
+@{*/
+/** TX FIFO Threshold Interrupt Flag. */
+#define USART_ISR_TXFT BIT27
+/** RX FIFO Threshold Interrupt Flag. */
+#define USART_ISR_RXFT BIT26
+/** Transmission Complete before Guard Time Interrupt Flag. */
+#define USART_ISR_TCBGT BIT25
+/** RX FIFO Full Flag. */
+#define USART_ISR_RXFF BIT24
+/** TX FIFO Empty Flag. */
+#define USART_ISR_TXFE BIT23
+/** SPI Slave Underrun Flag. */
```

Comment:
uart, not spi, same a little below?

Reply:
![image](https://user-images.githubusercontent.com/2933763/70837024-33578c00-1dd0-11ea-90b7-a9c1593c2fb7.png)